test/controllers/users_controller_test.rb

```diff
@@ -402,6 +402,16 @@ class UsersControllerTest < ActionController::TestCase
     assert_equal user.mail, mail.to[0]
   end
 
+  test "test email should be delivered even when user has read only role" do
+    user = users(:one)
+    user.roles = [Role.find_by_name('Viewer')]
+    User.current = user
```

Comment:
Please use FactoryGirl to create a user with only viewer permissions rather then modify the user in fixtures which may have unexpected side effects in other tests.